tests/test_custom_return_types.py

```diff
@@ -278,6 +278,254 @@ def test_hash_invalid_type(self):
             hash(res)
 
 
+class TestPathMapping(unittest.TestCase):
+
+    def setUp(self):
+        self.dag = retworkx.PyDAG()
+        node_a = self.dag.add_node('a')
+        self.dag.add_child(node_a, 'b', "Edgy")
+
+    def test__eq__match(self):
+        self.assertTrue(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) == {1: [0, 1]})
+
+    def test__eq__not_match_keys(self):
+        self.assertFalse(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) == {2: [0, 1]})
+
+    def test__eq__not_match_values(self):
+        self.assertFalse(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) == {1: [0, 2]})
+
+    def test__eq__different_length(self):
+        self.assertFalse(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) == {
+                1: [0, 1], 2: [0, 2]})
+
+    def test_eq__same_type(self):
+        self.assertEqual(retworkx.dijkstra_shortest_paths(self.dag, 0),
+                         retworkx.dijkstra_shortest_paths(self.dag, 0))
+
+    def test__eq__invalid_type(self):
+        self.assertFalse(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) == ['a', None])
+
+    def test__eq__invalid_inner_type(self):
+        self.assertFalse(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) == {0: {'a': None}})
+
+    def test__ne__match(self):
+        self.assertFalse(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) != {1: [0, 1]})
+
+    def test__ne__not_match(self):
+        self.assertTrue(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) != {2: [0, 1]})
+
+    def test__ne__not_match_values(self):
+        self.assertTrue(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) != {1: [0, 2]})
+
+    def test__ne__different_length(self):
+        self.assertTrue(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) != {
+                1: [0, 1], 2: [0, 2]})
+
+    def test__ne__invalid_type(self):
+        self.assertTrue(
+            retworkx.dijkstra_shortest_paths(self.dag, 0) != ['a', None])
+
+    def test__gt__not_implemented(self):
+        with self.assertRaises(NotImplementedError):
+            retworkx.dijkstra_shortest_paths(self.dag, 0) > {1: [0, 2]}
+
+    def test_deepcopy(self):
+        paths = retworkx.dijkstra_shortest_paths(self.dag, 0)
+        paths_copy = copy.deepcopy(paths)
```

Comment:
Should the `PathMapping` return a shallow copy as well, using `.copy()` just like a dict does?

Reply:
I hadn't thought of that (and honestly completely missed the `.copy()`method in the dict api). Thinking about it a bit I'm not sure it makes sense here though. Since `PathMapping` is read only a copy would be identical and couldn't diverge. The one way that would be different is if it returned a `dict` copy so you could change the object after copying. But that would be the same as just running `dict(path_mapping)`. So while the mapping types here are read-only I don't think we need `.copy()`.